types/massive/index.d.ts

```diff
@@ -42,194 +49,338 @@ declare namespace massive {
+            record: AnyObject,
+            options?: PersistenceInsertOptions &
+                PersistenceUpdateDocOptions &
+                ResultProcessingOptions
+        ): Promise<AnyObject>;
+
+        /** Save a document to the database. This function will create or replace the entire document body. */
+        saveDoc(doc: AnyObject): Promise<AnyObject>;
 
         /**
-         * Determine whether criteria represent a search by primary key.
-         * If a number or uuid are passed, it is assumed to be a primary key value; if an object, it must have only one key, which must specify the primary key column.
+         * Update a record. May be invoked with a complete record (including primary key), or with a criteria object and a map of fields to new values.
+         * Multi-row updates are only possible through the latter usage.
          */
-        isPkSearch(criteria?: object | string | number): boolean;
+        update(
+            criteria: string | number,
+            changes: AnyObject,
+            options?: PersistenceUpdateDocOptions & ResultProcessingOptions
+        ): Promise<AnyObject>;
+        update(
+            criteria: AnyObject,
+            changes: AnyObject,
+            options?: PersistenceUpdateDocOptions & ResultProcessingOptions
+        ): Promise<AnyObject[]>;
 
         /**
-         * Perform a full-text search on queryable fields. If options.document is true, looks in the document body fields instead of the table columns.
+         * Update a document, adding new information and changing existing information.
+         * This function can be used with any JSON field, not just document tables; however, only document tables can use criteria objects which directly reference document fields.
+         * If calling updateDoc with a criteria object for a non-document table, the criteria will be tested against the entire row (as opposed to the document body as it is for document tables).
+         * To test elements of the JSON field in a non-document table with a criteria object, use a JSON path string.
          */
-        search(plan: SearchDefinition, options?: GenericQueryOptions): Promise<any>;
+        updateDoc(
+            criteria: string | number | AnyObject,
+            changes: AnyObject,
+            options?: PersistenceUpdateDocOptions & ResultProcessingOptions
+        ): Promise<AnyObject>;
+    }
 
-        /** Shortcut to perform a full text search on a document table. */
-        searchDoc(plan: SearchDefinition, options?: GenericQueryOptions): Promise<any>;
+    class Sequence {
+        /** A database sequence. */
+        constructor(db: Database, name: string, schema: string);
 
-        /** Run a query with a raw SQL predicate, eg: db.mytable.where('id=$1', [123]).then(...); */
-        where(conditions: string, params?: any, options?: GenericQueryOptions): Promise<any>;
+        /** Get the last value the sequence returned. */
+        lastValue(): Promise<any>;
```

Comment:
The result of `lastValue` and `nextValue` will always be numeric, although since Postgres uses larger numbers than JavaScript can support it'll be stringified unless you do some surgery with node-pg-types.